Objects/stringlib/fastsearch.h

```diff
@@ -740,14 +740,31 @@ static inline Py_ssize_t
 STRINGLIB(count_char)(const STRINGLIB_CHAR *s, Py_ssize_t n,
                       const STRINGLIB_CHAR p0, Py_ssize_t maxcount)
 {
-    Py_ssize_t i, count = 0;
-    for (i = 0; i < n; i++) {
-        if (s[i] == p0) {
-            count++;
-            if (count == maxcount) {
-                return maxcount;
+    Py_ssize_t count = 0;
+    const STRINGLIB_CHAR *restrict cursor = s;
+    const STRINGLIB_CHAR *end_ptr = s + n;
+    const STRINGLIB_CHAR *unroll_end_ptr = end_ptr - 31;
+    /* By unrolling in chunks of 32, the compiler can auto vectorize, resulting
```

Comment:
Is 32 optimal for *any* supported architecture? or is it possible to use 64-bit chunks for 64-bit architecture?

Reply:
These are byte-chunks, not bitchunks. ARM64 and x86-64 have 16-byte (128-bit) vectors by default. Clang and GCC are able to use these properly. On other architectures the loop is simply unrolled. 

MSVC does not unroll the loop however I see. That might have an impact on performance. I should test that.

Reply:
> These are byte-chunks, not bitchunks

Oh yes, sorry (well, my question remains the same actually).

Reply:
Actually, I was wondering whether we could have a macro defining the correct constant to use depending on the architecture. That way, it could be more or less optimized per architecture.

Reply:
That's a good idea. Let's see what happens when I get to the windows benchmarking. MSVC does not unroll the inner loop at all, so the performance is potentially going to be very poor. I was also thinking of enclosing the unrolled loop in compile guards and only allow it for architectures that are known to perform better this way. Anyway, I will get to that after some benchmarks in the coming days.

Reply:
Alternatively, for MSVC, you could unroll the loop manually actually. While I may understand that it's maybe an overkill, it might be worthwhile I'd say.